docs/dex-blog-post.md

```diff
@@ -157,29 +83,93 @@ The seller contract guarantees that the seller box can be spent:
+{
+  pkA || {
+    val tokens = OUTPUTS(0).tokens
+    val tokenDataCorrect = tokens.nonEmpty &&
+      tokens(0)._1 == tokenId &&
+      tokens(0)._2 >= tokenAmount
+    val knownId = OUTPUTS(0).R4[Coll[Byte]].get == SELF.id
+    tokenDataCorrect &&
+    OUTPUTS(0).propositionBytes == pkA.propBytes &&
+    knownId
+  }
+}
+```  
+
+The buyer contract guarantees that the buyer box can be spent:
+1) by buyer itself, which is the way for buyer to [cancel the order](#canceling-the-orders)
+2) by a _swap transaction_ created by Matcher in which _buyer box_ is spent together (i.e atomically) 
+with the matched _seller box_ (see [sell tokens](#sell-tokens)).
+
+The following command can be used to create new _buy order_ to buy tokens:
+```
+Command Name:	dex:BuyOrder
+Usage Syntax:	ergo-tool dex:BuyOrder <wallet file> <ergAmount> <tokenId> <tokenAmount>, <dexFee>
+Description:	put a token buyer order with given <tokenId> and <tokenAmount> to buy at given <ergPrice> price with <dexFee> as a reward for anyone who matches this order with a seller, with wallet's address to be used for withdrawal
+ with the given <wallet file> to sign transaction (requests storage password)
+Doc page:	https://aslesarenko.github.io/ergo-tool/api/org/ergoplatform/appkit/ergotool/dex/CreateBuyOrderCmd.html
+```
+
+Here is an example of using `dex:BuyOrder` to submit the order to the blockchain - https://gist.github.com/greenhat/0c75738edadb9870a2cfb492d6069a57
+
+## List My Orders 
+
+To show your outstanding buy/sell orders (that use your public key in their contracts) use `dex:ListMyOrders` command:
+```
+Command Name:	dex:ListMyOrders
+Usage Syntax:	ergo-tool dex:ListMyOrders <storageFile>
+Description:	show buy and sell orders created from the address of this wallet
+Doc page:	https://aslesarenko.github.io/ergo-tool/api/org/ergoplatform/appkit/ergotool/dex/ListMyOrdersCmd.html
+```
+
+Here is an example of using `dex:ListMyOrders` - https://gist.github.com/greenhat/6b1b2f7be1279de49e33045b3fac6f81
+
 ## Cancel Order
 
+To cancel your buy/sell order you need to "spend" the box of the order by sending its assets (coins and/or tokens)
+to your own address.
+The following command can be used to spend your order box and send you the assets:
+```
+Command Name:	dex:CancelOrder
+Usage Syntax:	ergo-tool dex:CancelOrder <wallet file> <orderBoxId>
+Description:	claim an unspent buy/sell order (by <orderBoxId>) and sends the ERGs/tokens to the address of this wallet (requests storage password)
+Doc page:	https://aslesarenko.github.io/ergo-tool/api/org/ergoplatform/appkit/ergotool/dex/CancelOrderCmd.html
+```
+
+Here is an example of using `dex:CancelOrder` - https://gist.github.com/greenhat/6c70999c763a70a7253170d33127e9da
+
+## Overview
+
+To get an overview of how contracts and boxes play in transactions check out an Ergo playground with scenarios for the DEX commands - [Run in Scastie](https://scastie.scala-lang.org/greenhat/T2jSEv11QcWpXX1XrcHUdw/49)
+
```

Comment:
This deserves it own post. 
I mentioned this in `To recap` as 
"In the next posts we are going look under the hood and see how to implement new commands of
ErgoTool, stay tuned!"
So the development `under the hood` staff can be shown in a separate focused post.

Reply:
Removed.